aya/src/util.rs

```diff
@@ -88,29 +88,48 @@ impl KernelVersion {
         (u32::from(major) << 16) + (u32::from(minor) << 8) + u32::from(patch)
     }
 
-    // This is ported from https://github.com/torvalds/linux/blob/3f01e9f/tools/lib/bpf/libbpf_probes.c#L21-L101.
-
+    // This (L92-L133) is ported from https://github.com/torvalds/linux/blob/3f01e9f/tools/lib/bpf/libbpf_probes.c#L21-L101.
     fn get_ubuntu_kernel_version() -> Result<Option<Self>, CurrentKernelVersionError> {
+        if let Some(content) = Self::read_ubuntu_kernel_version_file()? {
+            if let Some(v) = Self::parse_ubuntu_kernel_version(&content).ok().flatten() {
```

Comment:
this can just be `if let Ok(Some(v)) = ...` instead of .ok().flatten()

Reply:
hope this works